clients/TypeScript/packages/client/src/LocalTxMonitor/LocalTxMonitorClient.ts

```diff
@@ -0,0 +1,195 @@
+ *
+ * @category LocalTxMonitor
+ **/
+export interface LocalTxMonitorClient {
+    context: InteractionContext
+    awaitAcquire: (args?: {}) => Promise<AwaitAcquired>
+    hasTx: (id: TxId) => Promise<boolean>
+    nextTx: (args?: { fields?: "all" }) => Promise<TxId | TxAlonzo | Null>
+    sizeAndCapacity: (args?: {}) => Promise<MempoolSizeAndCapacity>
+    releaseMempool: (args?: {}) => Promise<string>
+    shutdown: () => Promise<void>
+}
+
+/** @Internal */
+const matchAwaitAcquire = (data: string) => {
+    const response = safeJSON.parse(data) as Ogmios['AwaitAcquireResponse']
+    if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'AwaitAcquire') {
+        return null
+    }
+    return response
+}
+
+/** @Internal */
+const matchHasTx = (data: string) => {
+    const response = safeJSON.parse(data) as Ogmios['HasTxResponse']
+    if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'HasTx') {
+        return null
+    }
+    return response
+}
+
+/** @Internal */
+const matchNextTx = (data: string) => {
+    const response = safeJSON.parse(data) as Ogmios['NextTxResponse']
+    if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'NextTx') {
+        return null
+    }
+    return response
+}
+
+/** @Internal */
+const matchSizeAndCapacity = (data: string) => {
+    const response = safeJSON.parse(data) as Ogmios['SizeAndCapacityResponse']
+    if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'SizeAndCapacity') {
+        return null
+    }
+    return response
+}
+
+/** @Internal */
+const matchReleaseMempool = (data: string) => {
+    const response = safeJSON.parse(data) as Ogmios['ReleaseMempoolResponse']
+    if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'ReleaseMempool') {
+        return null
+    }
+    return response
+}
+
+/**
+ * Create a client for inspect the node’s local mempool.
+ *
+ * @category Constructor
+ **/
+export const createLocalTxMonitorClient = async (
+    context: InteractionContext
+): Promise<LocalTxMonitorClient> => {
+    const { socket } = context
+
+    const awaitAcquireResponse = eventEmitterToGenerator(socket, 'message', matchAwaitAcquire)() as
+        AsyncGenerator<Ogmios['AwaitAcquireResponse']>
+
+    const hasTxResponse = eventEmitterToGenerator(socket, 'message', matchHasTx)() as
+        AsyncGenerator<Ogmios['HasTxResponse']>
+
+    const nextTxResponse = eventEmitterToGenerator(socket, 'message', matchNextTx)() as
+        AsyncGenerator<Ogmios['NextTxResponse']>
+
+    const sizeAndCapacityResponse = eventEmitterToGenerator(socket, 'message', matchSizeAndCapacity)() as
+        AsyncGenerator<Ogmios['SizeAndCapacityResponse']>
+
+    const releaseMempoolResponse = eventEmitterToGenerator(socket, 'message', matchReleaseMempool)() as
+        AsyncGenerator<Ogmios['ReleaseMempoolResponse']>
```

Comment:
That's one way of doing it :sweat_smile: ... Intuitively, I would have gone for a single generator matching all the messages. This could reduce a bit the boilerplate here I reckon and would be fine because the server handles messages sequentially behind the scene :thinking: 

I know I did use two event queues in the TxSubmission protocol but in principle, there was no need and I guess I was didn't do this thought experiments.